### PR TITLE
WIP: commands: add submit results

### PR DESCRIPTION
**Pull request by mwasilew**
Signed-off-by: Milosz Wasilewski <milosz.wasilewski@linaro.org>

### CODE REVIEW

File: squad_client/commands/submit.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        parser.add_argument(
+            "--metrics",
+            help="File with metrics(benchmarsk) to submit. Max 5MB. JSON and YAML formats are supported",
+        )
+        parser.add_argument(
+            "--metadata",
+            help="File with metadata to submit. Max 5MB. JSON and YAML formats are supported",
+        )
+        parser.add_argument(
+            "--attachments",
```
Comment (chaws) [inline squad_client/commands/submit.py:38]: I get `TypeError: 'NoneType' object is not iterable` when not passing attachments, perhaps `default=[]` would fix it? I guess default is `None`

File: squad_client/commands/submit.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            if not self.__check_file(filename):
+                return False
+
+        if results_dict:
+            # check dictionary correctness
+            for key, value in iter(results_dict.items()):
+                if type(key) is not str:
+                    logger.error("Non-string key detected")
+                    return False
+                if type(value) is not str or type(value) is not dict:
```
Comment (chaws) [inline squad_client/commands/submit.py:127]: Did you mean `and`? It's falling in that condition `if value is str or value is dict` :)

**Comment by chaws** [comment]
I checked your branch locally and rebased it against https://github.com/Linaro/squad-client/pull/29 and had to fix two minor issues within shortcuts.py and get this command working:

```
./manage.py \
    --squad-host http://localhost:8000 \
    --squad-token cbcff1bd1947c5ff46284d1f0de9f1ab126ceb05 \
    submit \
        --group dummygroup \
        --project dummyproject \
        --build dummybuild \
        --environment dummyenvironment \
        --result-name lala \
        --result-value pass
```

I'll test it some more

**Comment by mwasilew** [comment]
Thanks @chaws, much appreciated. I'll fix the issues you pointed tomorrow.

File: squad_client/commands/submit.py (hunk 2 above)
Comment (mwasilew) [reply squad_client/commands/submit.py]: I changed it a bit so it's cleaner.

File: squad_client/commands/submit.py (hunk 1 above)
Comment (mwasilew) [reply squad_client/commands/submit.py:38]: ack, done.

**Comment by chaws** [comment]
Looks good, there are some other minor bugs in shortcuts, I'll merge this now and open a new PR with eventual bug fixes

**Pull request merged**